internal/log/zap.go

```diff
@@ -0,0 +1,25 @@
+package log
```

Comment:
thank you SO much for starting on this :) 

This might not be possible, but what do you think of us providing an Interface that people have to satisfy (and not including zap)

I just don't want to make people import/vendor another library (zap) if they don't have to

Reply:
The interface is in log.go. We can move the zap implementation to another package so it doesn't have to be imported all the time.

Reply:
awesome!

My only concern is just if someone vendors/builds pion. I don't want us to be broken/have our size balloon from Zap (I am fine with this dependency though, logging is a solved problem and just happy with what is popular)

Reply:
Yea, that's what I meant. We can move the zap.go file to another package so it's no longer a dependency. A user of the lib can optionally use it if they want or roll their own implementation of the log.Logger interface.